packages/insomnia-app/app/ui/components/wrapper-home.tsx

```diff
@@ -4,63 +4,156 @@ import { autoBindMethodsForReact } from 'class-autobind-decorator';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
+import { unreachableCase } from 'ts-assert-unreachable';
 
 import { parseApiSpec, ParsedApiSpec } from '../../common/api-specs';
 import {
   AUTOBIND_CFG,
+  SpaceSortOrder,
 } from '../../common/constants';
 import { hotKeyRefs } from '../../common/hotkeys';
 import { executeHotKey } from '../../common/hotkeys-listener';
-import { fuzzyMatchAll, isNotNullOrUndefined } from '../../common/misc';
-import { descendingNumberSort } from '../../common/sorting';
+import { isNotNullOrUndefined } from '../../common/misc';
+import { descendingNumberSort, sortMethodMap } from '../../common/sorting';
 import { strings } from '../../common/strings';
+import { ApiSpec } from '../../models/api-spec';
 import { isRemoteSpace } from '../../models/space';
 import { isDesign, Workspace, WorkspaceScopeKeys } from '../../models/workspace';
+import { WorkspaceMeta } from '../../models/workspace-meta';
 import { MemClient } from '../../sync/git/mem-client';
 import { initializeLocalProjectAndMarkForSync } from '../../sync/vcs/initialize-project';
 import coreLogo from '../images/insomnia-core-logo.png';
 import { cloneGitRepository } from '../redux/modules/git';
+import { setSpaceSortOrder } from '../redux/modules/global';
 import { ForceToWorkspace } from '../redux/modules/helpers';
 import { importClipBoard, importFile, importUri } from '../redux/modules/import';
 import { activateWorkspace, createWorkspace } from '../redux/modules/workspace';
-import Highlight from './base/highlight';
+import { selectSpaceSortOrder } from '../redux/selectors';
 import SettingsButton from './buttons/settings-button';
 import AccountDropdown from './dropdowns/account-dropdown';
 import { RemoteWorkspacesDropdown } from './dropdowns/remote-workspaces-dropdown';
 import { SpaceDropdown } from './dropdowns/space-dropdown';
-import { WorkspaceCardDropdown } from './dropdowns/workspace-card-dropdown';
+import OrderFilterDropdown from './dropdowns/space-sort-dropdown';
 import KeydownBinder from './keydown-binder';
 import { showPrompt } from './modals';
 import Notice from './notice';
 import PageLayout from './page-layout';
-import TimeFromNow from './time-from-now';
-import type {
-  WrapperProps,
-} from './wrapper';
-
-interface RenderedCard {
-  card: ReactNode;
-  lastModifiedTimestamp?: number | null;
-}
+import WorkspaceCard, { WorkspaceCardProps } from './workspace-card';
+import type { WrapperProps } from './wrapper';
 
-interface Props extends ReturnType<typeof mapDispatchToProps> {
+interface Props
+  extends ReturnType<typeof mapDispatchToProps>,
+    ReturnType<typeof mapStateToProps> {
   wrapperProps: WrapperProps;
 }
 
 interface State {
   filter: string;
 }
 
+function orderSpaceCards(orderBy: SpaceSortOrder) {
+  return (cardA: Pick<WorkspaceCardProps, 'workspace' | 'lastModifiedTimestamp'>, cardB: Pick<WorkspaceCardProps, 'workspace' | 'lastModifiedTimestamp'>) => {
```

Comment:
What is the rationale behind making these props `Pick<WorkspaceCardProps, ...>` instead of the entire `WorkspaceCardProps` interface?

Reply:
The WorspaceCard also accepts the `onSelect, activeSpace, filter` props which are required and not available on the orderSpaceCards fn. We could also make another type e.g.

```
interface SortableWorkspaceCard {
  workspace: Workspace;
  lastModifiedTimestamp: number;
}
``` 

and change it like that:

```suggestion
  return (cardA: SortableWorkspaceCard, cardB: SortableWorkspaceCard) => {
```

Reply:
Ah I understand. That's fine for the scope of this PR I think though this card logic could use another sweep further down the track. It just feels more complex than it realistically needs to be (not as a result of this PR, just the original implementation to begin with)